x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/AbstractBuilder.java

```diff
@@ -183,6 +184,48 @@ public static String unquoteString(Source source) {
         return resultString.toString();
     }
 
+    private static String handleUnicodeChars(Source source, String text) {
+        StringBuilder sb = new StringBuilder();
+
+        int startIdx = 0;
+        int endIdx = 0;
+        int idx = text.indexOf("\\u");
+        while (idx >= 0) {
+            String unicodeSequence;
+            if (text.charAt(idx + 2) == '{') {
+                endIdx = text.indexOf("}", idx + 1) + 1;
+                unicodeSequence = text.substring(idx + 3, endIdx - 1);
+                int length = unicodeSequence.length();
+                if (length < 2 || length > 8) {
+                    throw new ParsingException(source, "Unicode sequence in curly braces should use [2-8] hex digits, [{}] has [{}]",
+                            text.substring(idx, endIdx), length);
+                }
+            } else {
+                endIdx = idx + 6;
+                unicodeSequence = text.substring(idx + 2, endIdx);
+            }
+            sb.append(text, startIdx, idx).append(hexToUnicode(source, unicodeSequence));
+            idx = text.indexOf("\\u", endIdx);
```

Comment:
Personal preference (so very optional): this could be moved last in the loop (as it's the control value). Alternatively, the loop could be:
```
for (int idx = text.indexOf("\\u"); idx >= 0; idx = text.indexOf("\\u", endIdx)) {
```

Reply:
No worries, might be a bit clearer, and I also agree to use for loops if possible instead of while.